.clang-format.full

```diff
@@ -0,0 +1,115 @@
+---
+BasedOnStyle: LLVM
+Language: Cpp
+Standard: Auto
+
+IndentWidth: 2
+UseTab: Never
+ColumnLimit: 100
+
+ObjCBlockIndentWidth: 2
+
+MaxEmptyLinesToKeep: 1
+ReflowComments: true
+SortIncludes: CaseInsensitive
+SortUsingDeclarations: true
+
+AccessModifierOffset: -1
+AlignAfterOpenBracket: Align
+AlignConsecutiveMacros: true
+
+BinPackArguments: false
+BinPackParameters: false
+
+DerivePointerAlignment: false
+PointerAlignment: Right
+ReferenceAlignment: Left
+
+AllowShortBlocksOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLambdasOnASingleLine: Inline
+AllowShortLoopsOnASingleLine: false
+AllowShortEnumsOnASingleLine: false
+
+BreakConstructorInitializers: AfterColon
+
+BreakBeforeBraces: Custom
+BraceWrapping:
+  AfterClass: true
+  AfterControlStatement: false
+  AfterEnum: true
+  AfterExternBlock: true
+  AfterFunction: true
+  AfterNamespace: true
+  AfterObjCDeclaration: true
+  AfterStruct: true
+  AfterUnion: true
+  BeforeCatch: false
+  BeforeElse: false
```

Comment:
Most of our current code conforms to these, I believe:
```suggestion
  BeforeCatch: true
  BeforeElse: true
```

Reply:
And I would not mind a `BeforeWhile: true`

Reply:
Right. I did notice that and I changed it on purpose. I think `else` and `catch` should belong to the corresponding `if` and `try` in cases of multiple consecutive blocks. It's a very subjective thing though.

Keeping it as it is of course simplifies our formatting passes and the issues that will come because of it.